File: app/src/main/java/com/example/deltatask3/adapters/ItemLocationAdapter.java
package com.example.deltatask3.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.deltatask3.R;
import com.example.deltatask3.databinding.PokemonRowBinding;
import com.example.deltatask3.utils.ItemLocation;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import static com.example.deltatask3.UtilsKt.firstLetterToUppercase;

public class ItemLocationAdapter extends RecyclerView.Adapter<ItemLocationAdapter.ItemLocationViewHolder> {

    private ArrayList<ItemLocation> itemLocations;

    public void setItemLocations(ArrayList<ItemLocation> itemLocations) {
        this.itemLocations = itemLocations;
    }

    public static class ItemLocationViewHolder extends RecyclerView.ViewHolder {

        PokemonRowBinding binding;

        public ItemLocationViewHolder(@NonNull PokemonRowBinding pokemonRowBinding) {
            super(pokemonRowBinding.getRoot());
            binding = pokemonRowBinding;
        }
    }

    @NonNull
    @Override
    public ItemLocationViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        PokemonRowBinding binding = PokemonRowBinding.inflate(layoutInflater, parent, false);
        return new ItemLocationViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemLocationViewHolder holder, int position) {
        ItemLocation itemLocation = itemLocations.get(position);
        if (itemLocation == null) return;

        if (itemLocation.getSprite() != null) {
            holder.binding.ivPokemon.setVisibility(View.VISIBLE);
            String urlSprite = itemLocation.getSprite().getDefault_sprite();
            if (urlSprite != null && !urlSprite.isEmpty())
                Picasso.get().load(urlSprite).placeholder(R.drawable.placeholder_image).into(holder.binding.ivPokemon);
        } else
            holder.binding.ivPokemon.setVisibility(View.INVISIBLE);
        holder.binding.tvId.setText(String.valueOf(itemLocation.getId()));
        holder.binding.tvName.setText(firstLetterToUppercase(itemLocation.getName()));
    }

    @Override
    public int getItemCount() {
        return itemLocations.size();
    }
}

File: app/src/main/java/com/example/deltatask3/adapters/ItemLocationAdapter.kt
package com.example.deltatask3.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.deltatask3.R
import com.example.deltatask3.adapters.ItemLocationAdapter.ItemLocationViewHolder
import com.example.deltatask3.databinding.PokemonRowBinding
import com.example.deltatask3.firstLetterToUppercase
import com.example.deltatask3.models.ItemLocation
import com.squareup.picasso.Picasso

class ItemLocationAdapter(var itemLocations: ArrayList<ItemLocation>) : RecyclerView.Adapter<ItemLocationViewHolder>() {

    class ItemLocationViewHolder(var binding: PokemonRowBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemLocationViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = PokemonRowBinding.inflate(layoutInflater, parent, false)
        return ItemLocationViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ItemLocationViewHolder, position: Int) {
        val itemLocation = itemLocations[position]
        if (itemLocation.sprite != null) {
            holder.binding.ivPokemon.visibility = View.VISIBLE
            val urlSprite = itemLocation.sprite.default_sprite
            if (urlSprite != null && urlSprite.isNotEmpty())
                Picasso.get().load(urlSprite).placeholder(R.drawable.placeholder_image).into(holder.binding.ivPokemon)
        } else holder.binding.ivPokemon.visibility = View.INVISIBLE
        holder.binding.tvId.text = itemLocation.id.toString()
        holder.binding.tvName.text = firstLetterToUppercase(itemLocation.name)
    }

    override fun getItemCount() = itemLocations.size
}